src/main/java/com/lists/LinkedList.java

```diff
@@ -0,0 +1,192 @@
+	 * @return <code>true<code> if list is empty, <code>false</code> otherwise.
+	 */
+	public boolean isEmpty(){
+		return N == 0;
+	}
+	
+	
+	/**
+	 * append specified item to the end of this list
+	 * @param item specifies the item to be added
+	 */   
+	public void add(Item item) {
+		Node currentNode = head;    // reference of head is passed in temporary Node variable
+		Node newNode = new Node(item);
+		while (currentNode.next != null) {
+			currentNode = currentNode.next;
+		}
+		currentNode.next = newNode;
+		N++;
+	}
+
+	/**
+	 * append the specified item to the specified index of the list
+	 * @param item specifies the item to be added
+	 * @param index specifies the index value at which specified item to be added in integer
+	 * @throws IndexOutOfBoundsException if the index is out of range 
+	 */
+	public void add(Item item, int index) throws IndexOutOfBoundsException{
+		if (index <= 0 || index > size()) {
+			throw new IndexOutOfBoundsException();
+		}
+		Node currentNode = head;	
+		Node newNode = new Node(item);
+		for (int i = 1; i < index && currentNode.next != null; i++) {
+			currentNode = currentNode.next;
+		}
+		newNode.next = currentNode.next;
+		currentNode.next = newNode;
+		N++;
+	}
+
+	/**
+	 * returns the item at specified position in the list  
+	 * @param index specifies the position from the list in integer
+	 * @return item in the list
+	 * @throws IndexOutOfBoundsException if the index is out of range 
+	 */
+	public Item get(int index) throws IndexOutOfBoundsException {
+		if (index <= 0 || index > size()) {
+			throw new IndexOutOfBoundsException();
+		}
+		Node temp = head.next;
+		for (int i = 1; i < index; i++) {
+			if (temp.next == null)
+				return null;
+			temp = temp.next;
+
+		}
+		return temp.item;
+	}
+	/**
+	 * 
+	 * remove the node at specified position in the list  
+	 * @param index specifies the position from the list in integer
+	 * @return true removed node from the list
+	 */
+	public boolean remove(int index) {
+		if (index <= 0 || index > size()) {
+			return false;
+		}
+		Node current = head;
+		for (int i = 1; i < index; i++) {
+			if (current.next == null)
+				return false;
+			current = current.next;
+		}
+		current.next = current.next.next;
+		N--;
+		return true;
+	}
+	
+	
+	/**
+	 * returns an iterator over the LinkedList.
+	 * 
+	 */
+	public Iterator<Item> iterator()  { return new LinkedListIterator(); }
+	
+	
+	/*
+	 * LinkedListIterator implements Iterator interface
+	 * to provide Iterable capabilities to the LinkedList
+	 * data structure.
+	 * 
+	 *
+	 */
+	private class LinkedListIterator implements Iterator<Item> {
+        private Node current = head;
+        public boolean hasNext()  { return current != null;                     }
+        public void remove()      { throw new UnsupportedOperationException();  }
+
+        public Item next() {
+            if (!hasNext()) throw new NoSuchElementException();
+            Item item = current.item;
```

Comment:
Yes, code will not work at this point. I have changed the implementation. Check full code